### PR TITLE
Add Cookie api (fixes #21)

### LINKED ISSUE #21: Add API for managing cookies
Not necessarily final, but we need something like this:

New API in Browser like...
```java
browser.cookies().add(new Cookie(/*....*/));
browser.cookies().get("cookie_name");
browser.cookies().deleteAll()
// etc.etc.etc.
```

So that means you need two new classes in addition to Browser API changes:
`CookieManager` (what `browser.cookies()` returns)
And a `Cookie` class itself, which should basically look very similar to the one in WebDriver (if licenses are compatible we should consider actually use that one exactly and attributing it to the Selenium team in the license header but I'm not sure of all the legal ramifications of this).

Then you'll need to implement this in darcy-webdriver.

Do this work in a new version 0.2.1-SNAPSHOT so we don't break darcy-webdriver 0.2.0-SNAPSHOT. Same for darcy-webdriver version (it should be the next in line which I think should also be 0.2.1-SNAPSHOT).

### PR DESCRIPTION
**Pull request by abocz**
Need to add javadocs and determine what to do for the licensing still

### CODE REVIEW

File: src/main/java/com/redhat/darcy/web/api/CookieManager.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.redhat.darcy.web.api;
+
+import com.redhat.darcy.web.Cookie;
+
+import java.util.Set;
+
+public interface CookieManager{
+    void add(Cookie cookie);
+    void delete(Cookie cookie);
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/api/CookieManager.java]: I think this should probably accept a name String, not a cookie object, although you could have a default implementation that accepted a Cookie and call the String version.

File: src/main/java/com/redhat/darcy/web/api/CookieManager.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    void deleteAll();
+    Set<Cookie> getAll();
+    Cookie get(Cookie cookie);
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/api/CookieManager.java]: Same as above.. to keep this terse I would name `getNamed` simply `get`, and on the `Cookie` version of it, implement it with `default` and just call out to the `get(String)` version.

**Comment by alechenninger** [comment]
Looks good so far. Like you said, let's get the Selenium attribution and license in there.

I was also thinking if we should make this use case simpler...

```java
browser.cookies().getAll().stream()
  .filter(filterCookiesInSomeWay())
  .forEach(c -> browser.cookies().delete(c));
```

That's pretty terse already though so it might be fine, but, I dunno, do you think this would be useful?

```java
browser.cookies()
    .deleteWhere(c -> c.getDomain().contains("some.string"));

Set<Cookies> cookies = browser.cookies()
    .getWhere(c -> c.getDomain().contains("some.string"));
```

File: src/main/java/com/redhat/darcy/web/Cookie.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+        if (domain != null && domain.contains(":")) {
+            throw new IllegalArgumentException("Domain should not contain a port: " + domain);
+        }
+    }
+
+    @Override
+    public String toString() {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("EEE LLL dd HH:mm:ss z yyyy")
+                .withZone(ZoneId.systemDefault());
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/Cookie.java]: Not sure if this is the correct way to handle time zones.

File: src/main/java/com/redhat/darcy/web/Cookie.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
+package com.redhat.darcy.web;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+public class Cookie {
+    private static final long serialVersionUID = 4115876353625612383L;
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/Cookie.java]: Has a serial version UID but is not Serializable?

**Comment by abocz** [comment]
I could see this being useful, but not necessary at the moment. 
```
browser.cookies()
    .deleteWhere(c -> c.getDomain().contains("some.string"));

Set<Cookies> cookies = browser.cookies()
    .getWhere(c -> c.getDomain().contains("some.string"));
```

Can we make it an issue for later?

File: src/main/java/com/redhat/darcy/web/Cookie.java (hunk 3 above)
Comment (abocz) [inline src/main/java/com/redhat/darcy/web/Cookie.java]: I didn't see any good solutions for determining the user's timezone.  This uses the JVM's timezone.  The other option would be to not print / ignore timezone information.

**Comment by alechenninger** [comment]
> Can we make it an issue for later?

Sounds good to me! It's already quite nice with Java8 streams. I'll take a look at both of your PR's later this weekend. Thank you!

File: src/main/java/com/redhat/darcy/web/Cookie.java [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+    public LocalDateTime getExpiry() {
+        return expiry;
+    }
+
+    private static String stripPort(String domain) {
+        return (domain == null) ? null : domain.split(":")[0];
+    }
+
+    public void validate() {
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/Cookie.java]: I wonder why they did not do this validation on construction of the cookie. Perhaps for comparison's sake it is useful to be able to create invalid cookies? Not sure if this is the best way to handle that. Something to think about potentially refactoring later.

**Comment by alechenninger** [comment]
Thanks!

**Pull request merged**